py/miniconf-mqtt/miniconf/__init__.py

```diff
@@ -1,4 +1,52 @@
 #!/usr/bin/python3
 """ Root Miniconf module file. """
+import asyncio
+import logging
+import time
+
+from typing import List
+
+from gmqtt import Client as MqttClient
+
 from .miniconf import Miniconf
 from .version import __version__
+
+async def get_devices(broker: str,
+                      prefix_filter: str = None,
+                      discovery_timeout: float = 0.1) -> List[str]:
+    """ Get a list of available Miniconf devices.
+
+    Args:
+        * `broker` - The broker to search for clients on.
+        * `prefix_filter` - An MQTT-specific topic filter for device prefixes.
+        * `discovery_timeout` - The duration to search for clients in seconds.
+
+    Returns:
+        A list of discovered client prefixes that match the provided filter.
+    """
+    discovered_devices = []
+
+    suffix = '/connected'
+
+    def handle_message(_client, topic, payload, _qos, _properties):
+        if not topic.endswith(suffix):
+            return
+
+        logging.debug('Got message from %s: %s', topic, payload)
+        if payload == b"1":
+            discovered_devices.append(topic[:-len(suffix)])
+
+
+    client = MqttClient(client_id='')
+    client.on_message = handle_message
+
+    await client.connect(broker)
+
+    if prefix_filter is None:
+        prefix_filter = '#'
+
+    client.subscribe(prefix_filter)
```

Comment:
The default here is because we still want to be able to identify any miniconf device, where there's no required prefix.

For stabilizer and booster, we indeed can just provide a filter of `dt/insara/application/+`, but otherwise, Miniconf doesn't require that.

Reply:
Perhaps this is a point where we should address MQTT topic prefixes instead. I think this could be resolved by moving all of the device `alive` topics to be `alive/<prefix>` and then bulk subscribe to the `alive/#` topic, but that's also forcing some MQTT semantics on end users. However, I'm not convinced that's such a bad thing

Reply:
Alternatively, we can more strictly define the required prefix to be `<org>/<app>/<id>`, enforce that in code, and then use the `+/+/+/alive` wildcard string

Reply:
I don't think it's sensible to support the case where you don't even know what prefix your devices have. On a busy mqtt broker that's just too much traffic, and the likelyhood of a random `#/alive` being somewhere is very high. Then you also don't know what settings they have and can't do anything. In practice you will always know from the application you are trying to access what the prefix is. Let's have the user specify a sufficiently fine prefix pattern (here in miniconf) and don't provide a default.

Reply:
The only annoying thing is that the '+' symbol is single-level only, so `dt/sinara/+` will not match `dt/sinara/dual-iir/aa-bb-cc-dd-ee-ff`

Reply:
`dt/sinara/+/+`
But again. With a wildcard application name, you have no idea about the settings or the format.